src/main/java/com/example/demo/controller/BookController.java

```diff
@@ -33,12 +34,13 @@ public BookDto getBookById(@PathVariable Long id) {
     }
 
     @PostMapping
-    public BookDto createBook(@RequestBody CreateBookRequestDto requestDto) {
+    public BookDto createBook(@RequestBody @Valid CreateBookRequestDto requestDto) {
         return bookService.save(requestDto);
     }
 
     @PutMapping("/{id}")
-    public BookDto updateBook(@PathVariable Long id, @RequestBody CreateBookRequestDto requestDto) {
+    public BookDto updateBook(
+            @PathVariable Long id, @RequestBody @Valid CreateBookRequestDto requestDto) {
```

Comment:
```suggestion
            @PathVariable @Positive Long id, @RequestBody @Valid CreateBookRequestDto requestDto) {
```
You could also add validation for id. If so don't forget to add `@Validated` under the controller